src/main/java/com/Autopark/infrastructure/config/impl/JavaConfig.java

```diff
@@ -0,0 +1,41 @@
+package com.Autopark.infrastructure.config.impl;
+
+import com.Autopark.infrastructure.core.Scanner;
+import lombok.AllArgsConstructor;
+
+import java.util.Map;
+import java.util.Set;
+
+@AllArgsConstructor
+public class JavaConfig implements Config {
+    private final Scanner scanner;
+    private final Map<Class<?>, Class<?>> interfaceToImplementation;
+
+    @Override
+    public <T> Class<? extends T> getImplementation(Class<T> target) {
+        Set<Class<? extends T>> implementation = scanner.getSubtypesOf(target);
+
+        if (implementation.size() != 1) {
+            if (!isImplementationPresent(target)) {
+                throw new RuntimeException("target interface has 0 or more than 1 implementation");
```

Comment:
Maybe better create own exception?

Reply:
Maybe, but in a Task as needed use RuntimeException.